### PR TITLE
OSDOCS-1142: Adding PodLifeTime strategy and additional parameter for …

### PR DESCRIPTION
**Pull request by bergerhoffer**
…RemoveDuplicates

https://issues.redhat.com/browse/OSDOCS-1142

Preview: https://osdocs-1142--ocpdocs.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-descheduler.html#nodes-descheduler-strategies_nodes-descheduler

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@damemi Can you review these updates? Especially the capitalization - I changed the parameters to upper (pascal) case since that's what the others currently do.

Preview: https://osdocs-1142--ocpdocs.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-descheduler.html#nodes-descheduler-strategies_nodes-descheduler

I'm going to add the namespace/priority in a separate PR once I can test it in our operator so that I can get the right settings.

File: modules/nodes-descheduler-strategies.adoc [hunk 1]
```diff
@@ -20,6 +20,8 @@ Duplicate Pods::
 The `RemoveDuplicates` strategy ensures that there is only one Pod associated with a ReplicaSet, ReplicationController, Deployment, or Job running on same node. If there are more, then those duplicate Pods are evicted for better spreading of Pods in a cluster.
 +
 This situation could occur after a node failure, when a Pod is moved to another node, leading to more than one Pod associated with a ReplicaSet, ReplicationController, Deployment, or Job on that node. After the failed node is ready again, this strategy evicts the duplicate Pod.
++
+This strategy has an optional parameter, `ExcludeOwnerKinds`, that allows you to specify a list of `Kind` types. If a Pod has any of these types listed as an `OwnerRef`, then that Pod is not considered for eviction.
```
Comment (codyhoag) [inline modules/nodes-descheduler-strategies.adoc:24]: omit `then` after the comma

>! if...then _conj_
Avoid using “then” to introduce an independent clause that follows an “if” clause. For example, write “If
you set a password, access is restricted,” not “If you set a password, then access is restricted.”

File: modules/nodes-descheduler-strategies.adoc [hunk 2]
```diff
@@ -42,3 +44,9 @@ The `RemovePodsHavingTooManyRestarts` strategy ensures that Pods that have been
 This situation could occur if a Pod is scheduled on a node that is unable to start it. For example, if the node is having network issues and is unable to mount a networked persistent volume, then the Pod should be evicted so that it can be scheduled on another node. Another example is if the Pod is crashlooping.
 +
 This strategy has two configurable parameters: `PodRestartThreshold` and `IncludingInitContainers`. If a Pod is restarted more than the configured `PodRestartThreshold` value, then the Pod is evicted. You can use the `IncludingInitContainers` parameter to specify whether restarts for Init Containers should be calculated into the `PodRestartThreshold` value.
+
+Pod life time::
+
+The `PodLifeTime` strategy evicts Pods that are too old.
++
+After a Pod reaches the age, in seconds, set by the `MaxPodLifeTimeSeconds` parameter, it should be evicted.
```
Comment (codyhoag) [inline modules/nodes-descheduler-strategies.adoc:52]: Can something stronger be used here other than `it should be evicted`? Maybe `it is evicted.`?

**Review by codyhoag — COMMENTED** [review]
Two nits - otherwise looks good!

File: modules/nodes-descheduler-strategies.adoc (hunk 1 above)
Comment (bergerhoffer) [reply modules/nodes-descheduler-strategies.adoc:24]: This is a tough habit for me to break! :laughing:

**Review by damemi — COMMENTED** [review]
/lgtm
Thanks @bergerhoffer !

**Comment by bergerhoffer** [comment]
Thanks @damemi!

@kasturinarra Can you please QE review this update?

Preview: https://osdocs-1142--ocpdocs.netlify.app/openshift-enterprise/latest/nodes/scheduling/nodes-descheduler.html#nodes-descheduler-strategies_nodes-descheduler

**Comment by kasturinarra** [comment]
@bergerhoffer these changes have not been tested by QE yet, can we review the docs once the changes are available downstream ? JIRA here is still inprogress.

[1] https://issues.redhat.com/browse/WRKLDS-167

**Comment by bergerhoffer** [comment]
@kasturinarra Yes that works! I was planning to have another PR for the rest of the changes (namespace and priority filtering) once those were available downstream too, so that's fine to wait to do it all together.

**Comment by kasturinarra** [comment]
@bergerhoffer thanks, just reviewed and looks good to me, but one suggestion. Can we add the new strategies in under this point , so that user will know how to use them ? For example: PodLifeTime, ExcludeOwnerKinds etc.

Specify one or more strategies in the spec.strategies section.

**Comment by bergerhoffer** [comment]
@kasturinarra I updated to add the new strategy and parameter to that example. Can you let me know if that's what you were looking for, and whether this is good to go now? Thanks!

**Comment by kasturinarra** [comment]
@bergerhoffer looks good to me, thanks for adding it !!

**Pull request merged**